gator.Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.17-alpine AS builder
+
+COPY . /tmp/gatekeeper
+
+RUN apk add --no-cache \
+    upx
+
+WORKDIR /tmp/gatekeeper/cmd/gator
+
+RUN go get -d -v
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="-w -s" -o /gator
+RUN upx --best --lzma /gator
+
+FROM scratch as build
```

Comment:
Can we use `distroless` as the base image for uniformity with the G8r image?

https://github.com/open-policy-agent/gatekeeper/blob/9056cbbd77c085f2d33b4e9a29e0c084d745a00b/Dockerfile#L5